src/dmd/hdrgen.d

```diff
@@ -3107,10 +3107,14 @@ private void sizeToBuffer(Expression e, OutBuffer* buf, HdrGenState* hgs)
 {
     if (e.type == Type.tsize_t)
     {
-        Expression ex = (e.op == TOK.cast_ ? (cast(CastExp)e).e1 : e);
+        auto ce = e.isCastExp();
+        Expression ex = ce ? ce.e1 : e;
         ex = ex.optimize(WANTvalue);
-        const dinteger_t uval = ex.op == TOK.int64 ? ex.toInteger() : cast(dinteger_t)-1;
-        if (cast(sinteger_t)uval >= 0)
+
```

Comment:
This change should be the subject a different PR. There is no mention about this in the title or description.